eth/handler.go

```
@@ -221,16 +221,14 @@ func (pm *ProtocolManager) handle(p *peer) error {
 		p.Log().Debug("Ethereum handshake failed", "err", err)
 		// if error is due to GenesisBlockMismatch, NetworkIdMismatch, or ProtocolVersionMismatch
 		// and if sql database handle is available, update node information
-		if pm.db != nil && statusWrapper.isValidIncompatibleStatus() {
-			pm.storeEthNodeInfo(p.ID(), &statusWrapper)
+		if statusWrapper.isValidIncompatibleStatus() {
```

Why remove this validation check? Is it an attempt to fail hard if `pm.db == nil`?

---

It is to allow node-finder to continue updating the cached info as well as logging the status contents, regardless of sql db connection. Both are done by `pm.storeEthNodeInfo`, so I moved the `pm.db == nil` check to where the sql db update occurs within the function (https://github.com/teamnsrg/go-ethereum/pull/24/files#diff-4931b431e7ed8b0ade7b963e75eaa1edR61).